src/main/resources/default/taglib/t/blobImageSoftRefField.html.pasta

```diff
@@ -130,5 +147,37 @@
             }
             return '/' + path.substring(0, path.lastIndexOf('/') + 1);
         }
+
+        function isImage(extension) {
+            return extension === '.jpg' || extension === '.jpeg' || extension === '.png';
+        }
+
+        function determineFileIcon(extension) {
+            if (extension === '.pdf') {
+                return 'fa-file-pdf';
+            }
+
+            if (extension === '.tif' || extension === '.tiff' || extension === '.bmp' || extension === '.svg') {
+                return 'fa-image';
+            }
+
+            if (extension === '.doc' || extension === '.docx') {
+                return 'fa-file-word';
+            }
+
+            if (extension === '.xls' || extension === '.xlsx') {
+                return 'fa-file-excel';
+            }
+
+            if (extension === '.ppt' || extension === '.pps' || extension === '.pptx' || extension === '.ppsx') {
+                return 'fa-file-powerpoint';
+            }
+
+            if (extension === '.zip' || extension === '.rar') {
+                return 'fa-file-zipper';
+            }
```

Comment:
Are those all in the free version of Font Awesome? We also have such icons of our own in MIO.